packages/components/organisms/f-checkout/src/components/Selector.vue

```diff
@@ -119,13 +120,17 @@ export default {
         },
 
         /**
-         * Initialise the first set of times so if the user proceeds with the preset delivery time
-         * we have access to the from/to times up front.
+         * If we have previously set a time then initialise the `selected dropdown item` with that value; if
+         * currently present in the Fulfilment times array, otherwise initialise the with first value in the
+         * Fulfilment times array.
+         * Note; we always need to pre-select a time so the user can always proceeds with a selected time.
          *
-         * @param times
+         * @param {array} times - An array of available fulfilment times
          */
         initFulfilmentTime (times) {
-            if (times.length && times[0].value) {
+            if (this.time && this.time.from && times.length && times.find(i => i.value === this.time.from)) {
+                this.selectedAvailableFulfilmentTime = this.time.from;
+            } else if (times.length && times[0].value) {
```

Comment:
This seems a bit more complicated than what it could be, I think. Is it possible to simplify this?
I haven't checked if the code below works, but maybe it does, and if not, maybe it can give some ideas as in how to potentially make it cleaner:

```
initFulfilmentTime (times = []) {
  const selectedTime = times.find(time => time.value === this.time.from) || 
                                          times.length && times[0].value;

  this.selectedAvailableFulfilmentTime = selectedTime;

  this.updateFulfilmentTime({
      from: selectedTime,
      to: selectedTime
  });
}